amazon/ion/lazy_type.py

```diff
@@ -0,0 +1,24 @@
+from . import simpleion
+from .simple_types import _IonNature, IonPyNull, IonPyList
+
+
+class IonPyLazyObj(_IonNature):
+    """
+        Representation of an IonPyObj that generated by lazily_parse.
+        IonNature had ion_type already but I put it here for test purpose.
+    """
+    ion_buffer = None
+    ion_type = None
+
+    def __init__(self, b, t, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.lazy_buffer = b
+        self.lazy_type = t
+
+    # Wake up the lazy object, return a real IonPyObj
+    # This might be helpful later, but is not used at all for now
+    def wake_up(self):
+        if self.lazy_buffer is None:
+            return IonPyNull()
+        else:
+            return simpleion.loads(self.lazy_buffer)
```

Comment:
Note that this won't work as-is, because `lazy_buffer` doesn't hold the IVM.

Reply:
Okay, added an exception instead